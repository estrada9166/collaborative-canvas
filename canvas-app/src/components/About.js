import React, { Component } from 'react';
import {
  Step,
  Stepper,
  StepLabel,
} from 'material-ui/Stepper';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import { connect } from 'react-redux';
import {openSignModal} from '../actionCreators/modalStatus'
import CreateAccountModal from './CreateAccountModal'


class About extends Component {
  constructor(props) {
    super(props);
    this.state = {
      finished: false,
      stepIndex: 0,
    };

    this.handleNext = this.handleNext.bind(this);
    this.handlePrev = this.handleNext.bind(this);
    //this.handleRestart = this.handleRestart.bind(this);
    this.jumpToState = this.jumpToState.bind(this);
    this.open = this.open.bind(this);
  }

  handleNext(){
    const {stepIndex} = this.state;
    this.setState({
      stepIndex: stepIndex + 1,
      finished: stepIndex >= 2
    })
  }

  handlePrev(){
    const {stepIndex} = this.state;
    if(stepIndex > 0) {
      this.setState({
        stepIndex: stepIndex - 1
      })
    }    
  }

  // handleRestart(e){
  //   e.preventDefault();
  //   this.setState({
  //     stepIndex: 0, 
  //     finished: false
  //   });
  // }

  jumpToState(id){
    this.setState({
      stepIndex: id
    })
  }

  getStepContent(stepIndex) {
    if(stepIndex === 0){
      return 'Create a new account...';
    } else if(stepIndex === 1){
      return 'Send the link to your friends';
    } else if(stepIndex === 2){
      return 'Take all the benefits of a collaborative draw';
    } else {
      return 'Start drawing now!!';
    }
  }

  open() {
    this.props.open();
  }


  render() {
    const {finished, stepIndex} = this.state;
    const contentStyle = {margin: '0 16px'};

    return (
      <div style={{width: '100%', maxWidth: 700, margin: 'auto'}}>
        <Stepper activeStep={stepIndex}>
          <Step onClick={() => this.jumpToState(0)}>
            <StepLabel>Create an account</StepLabel>
          </Step>
          <Step onClick={() => this.jumpToState(1)}>
            <StepLabel>Invite your friends</StepLabel>
          </Step>
          <Step onClick={() => this.jumpToState(2)}>
            <StepLabel>Draw!!</StepLabel>
          </Step>
        </Stepper>
        <div style={contentStyle}>
          {finished ? (
            <p>
              <a onClick={this.open} >
                Click here
              </a> to create account.
            </p>
          ) : (
            <div>
              <p>{this.getStepContent(stepIndex)}</p>
              <div style={{marginTop: 12}}>
                <FlatButton
                  label="Create account"
                  style={{marginRight: 12}}
                  onTouchTap={this.open}
                />
                <RaisedButton
                  label={stepIndex === 2 ? 'Finish' : 'Next'}
                  primary={true}
                  onTouchTap={this.handleNext}
                />
              </div>
            </div>
          )}
        </div>
        <CreateAccountModal />
      </div>
    );
  }
}

const mapDispatchToProps = {
  open: () => openSignModal()
}

export default connect(null, mapDispatchToProps)(About);