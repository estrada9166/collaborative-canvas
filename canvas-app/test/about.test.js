import React from 'react';
import About from '../src/components/About';
import store from '../src/store';
import { Provider } from 'react-redux';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import renderer from 'react-test-renderer';

test('hola', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MuiThemeProvider>
        <About />
      </MuiThemeProvider>
    </Provider>
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  console.log(tree.children)
})