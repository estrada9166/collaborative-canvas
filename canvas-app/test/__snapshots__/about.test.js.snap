// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hola 1`] = `
<div
  style={
    Object {
      "margin": "auto",
      "maxWidth": 700,
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "muiPrepared": true,
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "flex": "0 0 auto",
          "muiPrepared": true,
        }
      }
    >
      <span
        disabled={undefined}
        style={
          Object {
            "alignItems": "center",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "flex",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 14,
            "fontWeight": 500,
            "height": 72,
            "muiPrepared": true,
            "paddingLeft": 14,
            "paddingRight": 14,
          }
        }
      >
        <span
          style={
            Object {
              "muiPrepared": true,
              "paddingRight": 8,
            }
          }
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#00bcd4",
                "display": "block",
                "fill": "#00bcd4",
                "fontSize": 24,
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <text
              fill="#fff"
              fontSize="12"
              textAnchor="middle"
              x="12"
              y="16"
            >
              1
            </text>
          </svg>
        </span>
        Create an account
      </span>
    </div>
    <div
      style={
        Object {
          "flex": "1 1 auto",
          "muiPrepared": true,
        }
      }
    >
      <span
        style={
          Object {
            "borderColor": "#bdbdbd",
            "borderTopStyle": "solid",
            "borderTopWidth": 1,
            "display": "block",
            "marginLeft": -6,
            "muiPrepared": true,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "flex": "0 0 auto",
          "marginLeft": -6,
          "muiPrepared": true,
        }
      }
    >
      <span
        disabled={true}
        style={
          Object {
            "alignItems": "center",
            "color": "rgba(0, 0, 0, 0.26)",
            "cursor": "default",
            "display": "flex",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 14,
            "height": 72,
            "muiPrepared": true,
            "paddingLeft": 14,
            "paddingRight": 14,
          }
        }
      >
        <span
          style={
            Object {
              "muiPrepared": true,
              "paddingRight": 8,
            }
          }
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#9e9e9e",
                "display": "block",
                "fill": "#9e9e9e",
                "fontSize": 24,
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <text
              fill="#fff"
              fontSize="12"
              textAnchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        Invite your friends
      </span>
    </div>
    <div
      style={
        Object {
          "flex": "1 1 auto",
          "muiPrepared": true,
        }
      }
    >
      <span
        style={
          Object {
            "borderColor": "#bdbdbd",
            "borderTopStyle": "solid",
            "borderTopWidth": 1,
            "display": "block",
            "marginLeft": -6,
            "muiPrepared": true,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "flex": "0 0 auto",
          "marginLeft": -6,
          "muiPrepared": true,
        }
      }
    >
      <span
        disabled={true}
        style={
          Object {
            "alignItems": "center",
            "color": "rgba(0, 0, 0, 0.26)",
            "cursor": "default",
            "display": "flex",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 14,
            "height": 72,
            "muiPrepared": true,
            "paddingLeft": 14,
            "paddingRight": 14,
          }
        }
      >
        <span
          style={
            Object {
              "muiPrepared": true,
              "paddingRight": 8,
            }
          }
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#9e9e9e",
                "display": "block",
                "fill": "#9e9e9e",
                "fontSize": 24,
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <text
              fill="#fff"
              fontSize="12"
              textAnchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        Draw!!
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "margin": "0 16px",
      }
    }
  >
    <div>
      <p>
        Create a new account...
      </p>
      <div
        style={
          Object {
            "marginTop": 12,
          }
        }
      >
        <button
          disabled={false}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "marginRight": 12,
              "minWidth": 88,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "verticalAlign": null,
              "zIndex": 1,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <span
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "muiPrepared": true,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "verticalAlign": "middle",
                }
              }
            >
              Create account
            </span>
          </div>
        </button>
        <div
          className={undefined}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "#ffffff",
              "borderRadius": 2,
              "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                     0 1px 4px rgba(0, 0, 0, 0.12)",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "minWidth": 88,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "#00bcd4",
                "border": 10,
                "borderRadius": 2,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 36,
                "lineHeight": "36px",
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": "100%",
                "zIndex": 1,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <div
                style={
                  Object {
                    "backgroundColor": false,
                    "borderRadius": 2,
                    "height": 36,
                    "muiPrepared": true,
                    "top": 0,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 14,
                      "fontWeight": 500,
                      "letterSpacing": 0,
                      "margin": 0,
                      "muiPrepared": true,
                      "opacity": 1,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "position": "relative",
                      "textTransform": "uppercase",
                      "userSelect": "none",
                    }
                  }
                >
                  Next
                </span>
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
